glow/src/triangle/gradient.rs

```diff
@@ -0,0 +1,189 @@
+    ) {
+        update_transform(gl, self.program, transform);
+
+        if &self.uniform_data.current_gradient != gradient {
+            match gradient {
+                Gradient::Linear(linear) => {
+                    let gradient_start: [f32; 2] = (linear.start).into();
+                    let gradient_end: [f32; 2] = (linear.end).into();
+
+                    unsafe {
+                        gl.uniform_2_f32(
+                            Some(
+                                &self
+                                    .uniform_data
+                                    .uniform_locations
+                                    .gradient_start_location,
+                            ),
+                            gradient_start[0],
+                            gradient_start[1],
+                        );
+
+                        gl.uniform_2_f32(
+                            Some(
+                                &self
+                                    .uniform_data
+                                    .uniform_locations
+                                    .gradient_end_location,
+                            ),
+                            gradient_end[0],
+                            gradient_end[1],
+                        );
+
+                        gl.uniform_1_u32(
+                            Some(
+                                &self
+                                    .uniform_data
+                                    .uniform_locations
+                                    .color_stops_size_location,
+                            ),
+                            linear.color_stops.len() as u32,
+                        );
+
+                        for (index, stop) in
+                            linear.color_stops.iter().enumerate()
+                        {
+                            gl.uniform_1_f32(
+                                Some(
+                                    &self
+                                        .uniform_data
+                                        .uniform_locations
+                                        .color_stops_locations[index]
+                                        .offset,
+                                ),
+                                stop.offset,
+                            );
+
+                            gl.uniform_4_f32(
+                                Some(
+                                    &self
+                                        .uniform_data
+                                        .uniform_locations
+                                        .color_stops_locations[index]
+                                        .color,
+                                ),
+                                stop.color.r,
+                                stop.color.g,
+                                stop.color.b,
+                                stop.color.a,
+                            );
+                        }
+                    }
```

Comment:
I wonder if there is a way to avoid many memory transfers and instead perform a single one for all the stops?